### ISSUE TITLE
[Question] How does it perform?

### ISSUE THREAD

**Issue by AcidRaZor** [WolfgangOfner/MicroserviceDemo#11]
I've been looking for a simple answer before doing a PoC converting old background applications using a queue to a function app.

Does the function app trigger for messages in sequence, or does it trigger for all message coming into the queue?

Meaning, if I have 10 messages that come in, does it process all 10 messages in parallel, or does message 2 have to wait for message 1 to complete?

Apologies if asking this here is out of scope. Found your tutorial but it didn't really show multiple messages and the question has been nagging me a little :P

**Comment by WolfgangOfner** [comment]
The Azure Function is triggered every time a new message comes in. But this does not mean that you have 10 Functions running when you send 10 messages at the same time. Azure takes care of the scaling of your Functions. It is possible that Azure runs your Function 10 times or scales to more instances until all messages are processed.

You can check out https://docs.microsoft.com/en-us/azure/azure-functions/event-driven-scaling for more details about scaling and this article also explains how you can configure the scaling behaviour and its limitations.

**Comment by AcidRaZor** [comment]
Thanks, I just clicked through to the index of your site and saw its way mode in-depth. seems I should stop late-night Googling on my phone as I just saw/read https://programmingwithwolfgang.com/rabbitmq-in-an-asp-net-core-3-1-microservice :P 

Thanks again
